### PR TITLE
Fix the problem that in the Triple protocol, an error will be reported immediately when the service is exposed

### PR DESCRIPTION
**Pull request by xiaoheng1**
## What is the purpose of the change

Fix the problem that in the Triple protocol, an error will be reported immediately when the service is exposed

see more detail from #8124 

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by horizonzy — APPROVED** [review]
lgtm.

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java [hunk 1]
```diff
@@ -149,14 +154,17 @@ public void onGoaway(Channel channel) {
 
     public void onConnected(Channel channel) {
         this.channel.set(channel);
+        // This indicates that the connection is available.
+        this.connectedFuture.complete(new Object());
```
Comment (guohao) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java:158]: replact `new Object()` with const instance maybe better?

Comment (xiaoheng1) [reply dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java:158]: ok, i will modify it.

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java [hunk 2]
```diff
@@ -149,14 +155,16 @@ public void onGoaway(Channel channel) {
...
         channel.attr(CONNECTION).set(this);
         if (logger.isInfoEnabled()) {
             logger.info(String.format("%s connected ", this));
         }
     }
 
-    public void connectSync() {
-        this.initPromise.awaitUninterruptibly(this.connectTimeout);
+    public void connectSync() throws InterruptedException, ExecutionException, TimeoutException {
+        this.connectedFuture.get(this.connectTimeout, TimeUnit.MILLISECONDS);
```
Comment (horizonzy) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java:167]: Why change the origin modify. Is there some problem when use spin to check isAvailabel()?
If we use future to realize sync mechanism, the connectedFuture is stateful, if the connect close and then reconnect, the connectFuture can't gurantee sync mechanism.

Comment (xiaoheng1) [reply dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java:168]: Using CompletableFuture code looks more elegant. And I also think that connectedFuture should be reset when the channel is unavailable. I modify the code.

**Review by xiaoheng1 — COMMENTED** [review]
Using CompletableFuture code looks more elegant. And I also think that connectedFuture should be reset when the channel is unavailable. I modify the code.

**Review by horizonzy — APPROVED** [review]
LGTM.

**Review by guohao — APPROVED** [review]

**Pull request merged**